window-copy.c

```diff
@@ -3083,10 +3101,19 @@ window_copy_search(struct window_mode_entry *wme, int direction, int regex,
 	wrapflag = options_get_number(wp->window->options, "wrap-search");
 	cis = window_copy_is_lowercase(str);
 
-	if (direction)
+	/*
+	 * Behave according to mode-keys. If it is emacs, search forward leaves the
+	 * cursor after the match. If it is vi, the cursor remains at the beginning of
+	 * the match, regardless of direction.
+	 */
+	keys = options_get_number(wp->window->options, "mode-keys");
+	if (direction) {
+		if (keys == MODEKEY_VI)
+			window_copy_move_right(s, &fx, &fy, wrapflag);
```

Comment:
In order to fix what https://github.com/tmux/tmux/commit/df7fbcd7a5a775586522380a70530721d1f2c151 was fixing as well, I need to move right here the length of the current search result. 

@nicm is there a way for me to get that here?